### PR TITLE
Add sodiumoxide unmaintained

### LINKED ISSUE #1090: `sodiumoxide` is (semi-)unmaintained
See https://github.com/sodiumoxide/sodiumoxide/commit/5bb1dfd2578539b89ffb0cbea25f21f00cfb963e#commitcomment-58111537

I asked @dnaq on that issue about whether we should file an unmaintained crate advisory for it, because unlike the crates we typically file advisories for it sounds like security-related fixes will be included/released. I think we're both a bit unclear on if it makes sense here.

I am starting to lean towards filing an advisory just because `sodiumoxide` has historically been a go-to solution for Rust cryptography and is still widely used (1,500-2,000 downloads per day). I was [recommending `sodiumoxide` as far back as 2014](https://speakerdeck.com/tarcieri/thoughts-on-rust-cryptography?slide=84).

It would be nice to point people at alternatives, however I don't know what good alternatives are. The `rust_sodium` crate is unmaintained, for example. There are also placeholder `sodium` and `libsodium` crates.

I know @jedisct1 has a [fork of `libsodium-sys`](https://github.com/jedisct1/libsodium-sys-stable).

There are also various pure Rust crates that implement NaCl protocols, such as [`xsalsa20poly1305`](https://github.com/RustCrypto/AEADs/tree/master/xsalsa20poly1305), the @RustCrypto [nacl-compat](https://github.com/RustCrypto/nacl-compat) crates, and [`dryoc`](https://crates.io/crates/dryoc).

I think if we can come up with some actionable advice for what current users of `sodiumoxide` should migrate to, it would make sense to file an unmaintained crate advisory.

### PR DESCRIPTION
**Pull request by pinkforest**
Closes #1090 

We will need to tune the alternatives and put some more wording around probably

### CODE REVIEW

**Comment by pinkforest** [comment]
@kpp any feedback very welcome around wording e.g. alternatives what should we recommend etc.

**Review by tarcieri — APPROVED** [review]

File: crates/sodiumoxide/RUSTSEC-0000-0000.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+```
+
+# sodiumoxide is deprecated
+
+Alternatives may be found - not in any specific order:
+
+- [libsodium-sys-stable](https://crates.io/crates/libsodium-sys-stable)
+- [dryoc](https://crates.io/crates/dryoc)
+- [RustCrypto/nacl-compat](https://github.com/RustCrypto/nacl-compat)
+- [RustCrypto/xsalsa20poly1305](https://github.com/RustCrypto/AEADs/tree/master/xsalsa20poly1305)
```
Comment (tarcieri) [inline crates/sodiumoxide/RUSTSEC-0000-0000.md:21]:
```suggestion
- [RustCrypto/nacl-compat](https://github.com/RustCrypto/nacl-compat) (`crypto_box`, `crypto_kx`, `crypto_secretstream`)
- [RustCrypto/xsalsa20poly1305](https://github.com/RustCrypto/AEADs/tree/master/xsalsa20poly1305) (`crypto_secretbox`)
```

**Comment by tarcieri** [comment]
It would probably be good to list some Ed25519 libraries like `ed25519-compact `, `ed25519-dalek`, and *ring*

**Comment by 5225225** [comment]
I'm not aware of any alternatives, but I posted a link to this issue in the #cryptography-and-security channel of the community discord to see if any of them have any suggestions.

**Review by dbrgn — APPROVED** [review]
LGTM!

**Comment by kpp** [comment]
I think you provided a wide and extended list of alternatives. Thanks

**Pull request merged**